ecdf_vat_report/models/ecdf_vat_report.py

```diff
@@ -0,0 +1,588 @@
+        '''
+        return '1.1'
+
+    def get_interface(self):
+        '''
+        :returns: eCDF interface ID (provided by eCDF)
+        '''
+        return 'CODL7'
+
+    @api.multi
+    def get_matr_declarer(self):
+        '''
+        :returns: Luxemburg matricule of the company
+        :raises: UserError: if no matricule
+        '''
+        self.ensure_one()
+        matr = self.company_id.l10n_lu_matricule
+        if not matr:
+            raise UserError(_('The company has no matricule set.'))
+        return matr
+
+    @api.multi
+    def get_rcs_declarer(self):
+        '''
+        :returns: RCS number of the company, 7 characters
+        If no RCS number, default value 'NE' is returned
+        (RCS : 'Numéro de registre de Commerce et des Sociétés')
+        '''
+        self.ensure_one()
+        rcs = self.company_id.company_registry
+        return rcs if rcs else 'NE'
+
+    @api.multi
+    def get_vat_declarer(self):
+        '''
+        :returns: VAT number of the company, 8 characters
+        If no VAT number, default value 'NE' is returned
+        '''
+        self.ensure_one()
+        vat = self.company_id.vat
+        if vat and vat.startswith('LU'):
+            return vat[2:]
+        return 'NE'
+
+    @api.multi
+    def get_matr_agent(self):
+        '''
+        :returns: Agent matricule provided in the form
+        If no agent matricule provided, the company one is returned
+        '''
+        self.ensure_one()
+        if self.agent_id and self.agent_id.matr:
+            return self.agent_id.matr
+        return self.get_matr_declarer()
+
+    @api.multi
+    def get_rcs_agent(self):
+        '''
+        :returns: RCS number (Numéro de registre de Commerce et des Sociétés)\
+        provided in the form.
+        If no RCS number has been provided, the company one is returned
+        If no RCS number of the company, default value 'NE' is returned
+        '''
+        self.ensure_one()
+        if self.agent_id and self.agent_id.rcs:
+            return self.agent_id.rcs
+        return self.get_rcs_declarer()
+
+    @api.multi
+    def get_vat_agent(self):
+        '''
+        :returns: VAT number provided in the form. If no VAT number has been\
+        provided, the VAT number of the company is returned.
+        If no VAT number of the company, default value 'NE' is returned
+        '''
+        self.ensure_one()
+        if self.agent_id and self.agent_id.vat:
+            vat = self.agent_id.vat
+            if vat.startswith('LU'):
+                return vat[2:]
+        return self.get_vat_declarer()
+
+    @api.multi
+    def get_language(self):
+        '''
+        :returns: the selected language in the form. Values can be :
+                    - "FR" for french
+                    - "DE" for german
+                    - "EN" for english
+        '''
+        self.ensure_one()
+        return self.language
+
+    @api.multi
+    def get_mis_report_month(self, year='2019'):
```

Comment:
```suggestion
    def get_mis_report_for_month(self, year):
```

Reply:
I modify it into:
`def get_mis_report(self, year, period_type)`
This way we will be able to easily add the declaration type such as Quarterly and Yearly